tests/file_storage/test_inmemory_storage.py

```diff
@@ -49,6 +49,14 @@ def test_file_saving(self):
         self.storage.save("file.dat", ContentFile(b"test"))
         self.assertEqual(self.storage.open("file.dat", "rb").read(), b"test")
 
+    def test_file_save_without_name(self):
+        """
+        File storage extracts the filename from the content object if no
+        name is given explicitly.
+        """
+        self.storage.save(None, ContentFile("test", name="file.txt"))
+        self.assertEqual(self.storage.open("file.txt", "r").read(), "test")
```

Comment:
This test passes with the changes reverted (and so isn't a regression test)

Reply:
Thanks for the review

I thought I used `InMemoryFileNode` instead of its parent class. I just fixed it. The test now fails with changes reverted:
```
tox -e py3 -- file_storage.test_inmemory_storage

ERROR: test_file_save_without_name (file_storage.test_inmemory_storage.MemoryStorageIOTests)
File storage extracts the filename from the content object if no
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/home/lucas/repos/django/tests/file_storage/test_inmemory_storage.py", line 58, in test_file_save_without_name
    self.storage.save(None, InMemoryFileNode("test", name="file.txt"))
  File "/home/lucas/repos/django/django/core/files/storage/base.py", line 32, in save
    name = content.name
AttributeError: 'InMemoryFileNode' object has no attribute 'name'

----------------------------------------------------------------------
Ran 24 tests in 0.522s

FAILED (errors=1)
```